apps/ensadmin/src/lib/chains.ts

```diff
@@ -1,22 +1,22 @@
-import { type Datasource, type ENSDeploymentChain, ENSDeployments } from "@ensnode/ens-deployments";
+import { type L1Chain, getENSDeployment } from "@ensnode/ens-deployments";
 import { type Chain } from "viem";
 
 /**
- * Get the chain by ID based on the current ENSDeployment configuration.
+ * Get a chain object by ID within the context of a specific ENSDeployment (identified by L1Chain).
  *
- * @param ensDeploymentChain - the ENSDeployment chain to get the chain for
- * @param chainId the chain ID to get the chain for
- * @returns the chain
- * @throws if the chain ID is not supported for the ENSDeployment chain
+ * @param l1Chain - the L1Chain identifying which ENSDeployment to query within
+ * @param chainId the chain ID
+ * @returns the viem#Chain object
+ * @throws if the chain ID is not supported within the selected ENSDeployment
```

Comment:
```suggestion
 * @throws if no Datasources are defined for the chainId in relation to selected l1Chain
```

This seems more accurate based on how I understand the logic here.